cirq/vis/heatmap.py

```diff
@@ -52,220 +35,318 @@ def relative_luminance(color: np.ndarray) -> float:
     return vis_utils.relative_luminance(color)
 
 
-class Heatmap:
-    """Distribution of a value in 2D qubit lattice as a color map."""
+QubitTuple = Tuple[grid_qubit.GridQubit, ...]
+Point = NamedTuple('Point', [('x', SupportsFloat), ('y', SupportsFloat)])
+PolygonUnit = NamedTuple(
```

Comment:
use dataclasses instead of NamedTuples

Reply:
done. 

Any specific reason why dataclasses should be preferred over NamedTuples?

Reply:
You should avoid namedtuple unless you specifically want to define a tuple. It defines "too much" -- a Point class defined this way can be iterated over or serialized as a JSON array, and behaves unexpectedly with %-formatting. namedtuple is great for making existing code that uses tuples more readable, but new classes that don't need tuple-compatibility should try dataclasses.

Reply:
Thanks for the explanation.